src/app/shared/components/disk-offering/disk-offering-dialog/disk-offering-dialog.component.ts

```diff
@@ -46,6 +55,14 @@ export class DiskOfferingDialogComponent {
   public isSubmitButtonDisabled() {
     const isDiskOfferingNotSelected = !this.selectedDiskOffering;
     const isNoDiskOfferings = !this.diskOfferings.length;
-    return isDiskOfferingNotSelected || isNoDiskOfferings;
+    return this.resourcesLimitExceeded || isDiskOfferingNotSelected || isNoDiskOfferings;
+  }
+
+  private checkResourcesLimit() {
+    const diskSize = isCustomized(this.selectedDiskOffering)
```

Comment:
extract function `isResourceLimitExceeded`, which returns `Number(storageAvailable) < Number(diskSize);`

then you assign the result to `this.resourcesLimitExceeded` in `checkResourcesLimit`